### DISCUSSION TITLE
iTwin Web Viewer - endless compile loop

### DISCUSSION THREAD

**Discussion by Johannes-repos** [iTwin/itwinjs-core#5527] [Q&A]
Hello,
I am getting an endless compile loop when starting the app in local host.
Started from npx create-react-app@latest my-app --scripts-version @bentley/react-scripts --template @itwin/web-viewer

and added
npm i @itwin/grouping-mapping-widget
npm i @itwin/reports-config-widget-react

the app runs fine but in VS code the compile starts over and over again.

https://github.com/iTwin/itwinjs-core/assets/39602714/5e714c92-3fd8-44f1-8be9-0a364385a90d

Anything I need to add?

Thanks

**Comment by aruniverse** [comment]
Set the env var USING_NPM to true in your .env file

**Reply by Johannes-repos** [reply]
tried that but it didn't help. The app is working.

**Reply by aruniverse** [reply]
~My apologies, the right env var name is `IS_USING_NPM`. please set that to true in your .env~

Sorry, it actually is `USING_NPM`. PR where we introduced this change: https://github.com/imodeljs/create-react-app/pull/99

**Reply by Johannes-lab** [reply]
Hi again,
looks like the latest version  @itwin/grouping-mapping-widget 0.16.1 fixed the compile loop.

Thanks for looking into this.

**Comment by Paolo-lab** [comment]
Hi I solved your same problem using the react-script version "4.0.7" , what version do you have ?

**Reply by Johannes-lab** [reply]
I am using 5.0.4

**Comment by Paolo-lab** [comment]
I m using 4.0.7

try to downgrade to 4.0.7

**Reply by Johannes-lab** [reply]
Hi,
tried a downgrade but it gave me a new error:
Error: error:0308010C:digital envelope routines::unsupported

using this: https://stackoverflow.com/questions/69692842/error-message-error0308010cdigital-envelope-routinesunsupported#:~:text=In%20your%20package.json%3A%20change%20this%20line

sorted it.

On top of that I needed to add/update https://www.npmjs.com/package/@babel/plugin-proposal-private-property-in-object
Not entirely sure what that does but the app now runs without errors or the compile loop. 

To test things further, I proceeded to upgrade back to 5.0.4 but that actually brought back the compile loop.

Sticking with 4.0.7

Thanks
